python/tests/avro/test_resolver.py

```diff
@@ -0,0 +1,137 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import pytest
```

Comment:
I'd recommend taking a look at the test cases in Java, which are pretty thorough and can find a lot of weird cases: https://github.com/apache/iceberg/blob/bd9408499cc99876fc52ebe3c00c9e40bd0e66c2/data/src/test/java/org/apache/iceberg/data/TestReadProjection.java#L38

Reply:
I've been looking into this, and am happy to replicate the tests on the Python side. However, it depends on the `writeAndRead` method that requires a write path. Maybe an idea to postpone this until we have the write path (which I see already glooming in the not-so-distant future).